aiohttp/client_exceptions.py

```diff
@@ -238,17 +239,34 @@ class InvalidURL(ClientError, ValueError):
 
     # Derive from ValueError for backward compatibility
 
-    def __init__(self, url: Any) -> None:
+    def __init__(self, url: StrOrURL, description: Optional[str] = None) -> None:
         # The type of url is not yarl.URL because the exception can be raised
         # on URL(url) call
-        super().__init__(url)
+        if description:
+            super().__init__(url, description)
+        else:
+            super().__init__(url)
+
+    @property
+    def url(self) -> StrOrURL:
+        return self.args[0]  # type: ignore[no-any-return]
 
     @property
-    def url(self) -> Any:
-        return self.args[0]
+    def description(self) -> Optional[str]:
+        try:
+            return self.args[1]  # type: ignore[no-any-return]
+        except KeyError:
+            return None
 
     def __repr__(self) -> str:
-        return f"<{self.__class__.__name__} {self.url}>"
+        return f"<{self.__class__.__name__} {self}>"
+
+    def __str__(self) -> str:
+        return " - ".join(map(str, self.args))
+
+
+class InvalidRedirectURL(InvalidURL):
+    """Invalid redirect url error"""
```

Comment:
```suggestion
    """Invalid redirect URL error."""
```

Don't forget to add a long description below this too.